rocAL/include/parameters/parameter_simple.h

```diff
@@ -48,6 +74,8 @@ class SimpleParameter : public Parameter<T> {
 
    private:
     T _val;
+    std::vector<T> _array;
+    unsigned _batch_size;
```

Comment:
Do we need to keep it as batch size here?

The size can be anything, so in all these parameters better to have as something which denotes size of the array you have allocated. Please rename these

Reply:
Address this.